### PR TITLE
Add tutorial for hastebin

### PR DESCRIPTION
**Pull request by squidboylan**

### CODE REVIEW

**Comment by smaffulli** [comment]
Looks good to me.

File: source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+The first step to deploying Hastebin is to launch a server to run it on. For
+example in this tutorial, an Ubuntu Xenial server is used. Read `How to launch
+and manage servers with the DreamCompute dashboard
+<https://help.dreamhost.com/hc/en-us/articles/215912848-How-to-launch-and-manage-servers-with-the-DreamCompute-dashboard>`__
+for information on how to do this.
+You also need to expose port 8080 to incoming traffic,
+as that is blocked by default. Read `How to configure access and security for
+DreamCompute instances
+<https://help.dreamhost.com/hc/en-us/articles/215912838-How-to-configure-access-and-security-for-DreamCompute-instances>`__
+for information on how to do this
```
Comment (justinknierim) [inline source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst]: period at the end of sentence.  Sorry, nitpicky as heck!  ;)

File: source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+    Deploying Hastebin as a service (as done in this article) requires you to
+    have root permissions. In order to start a root shell type ``sudo su -``.
+
+Once you have your server up and running the next step is to install all of
+Hastebin's dependencies:
+
+.. code:: bash
+
+    apt-get install nodejs npm
```
Comment (justinknierim) [inline source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst]: Probably is not important, but should a newly setup instance have an "apt-get update" ran first to fetch latest versions and such?  Just a thought, may not need a change.

**Comment by justinknierim** [comment]
From the style guideline, there is some use of "you" in there, but besides that it looks good to me!

**Comment by squidboylan** [comment]
I dont think using "you" is against the syleguide, you might be thinking of using "we" which is against the styleguide?

File: source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+.. Note::
+
+    Deploying Hastebin as a service (as done in this article) requires you to
+    have root permissions. In order to start a root shell type ``sudo su -``.
+
+Once you have your server up and running the next step is to install all of
+Hastebin's dependencies:
+
+.. code:: bash
```
Comment (cjdh) [inline source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst]: Update command line format:

.. code-block:: console

    [server]$ apt-get update
    [server]$ apt-get install nodejs npm

File: source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+    apt-get update
+    apt-get install nodejs npm
+
+Next you must symlink /usr/bin/nodejs to /usr/bin/node because Hastebin will
+try to use that path. Most Linux distributions install nodejs in /usr/bin/node.
+This step is only necessary on Ubuntu servers since it doesn't install nodejs
+in /usr/bin/node because of another package.
+
+.. code:: bash
```
Comment (cjdh) [inline source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst]: Update command line format:

.. code-block:: console

    [server]$ ln -s /usr/bin/nodejs /usr/bin/node

File: source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+
+    ln -s /usr/bin/nodejs /usr/bin/node
+
+Installing Hastebin
+~~~~~~~~~~~~~~~~~~~
+
+Now that all the dependencies are installed the next step is to download
+Hastebin and run it. To clone Hastebin using git, run:
+
+.. code:: bash
```
Comment (cjdh) [inline source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst]: Update command line format:

.. code-block:: console

    [server]$ git clone https://github.com/seejohnrun/haste-server.git /srv/haste-server

File: source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+
+    git clone https://github.com/seejohnrun/haste-server.git /srv/haste-server
+
+Then to install the application, run:
+
+.. code:: bash
```
Comment (cjdh) [inline source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst]: Update command line format:

.. code-block:: console

    [server]$ cd /srv/haste-server
    [server]$ npm install

File: source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+    User=haste-server
+    Group=haste-server
+    ExecStart=/usr/bin/node server.js
+
+    [Install]
+    WantedBy=multi-user.target
+
+Next we need to create the user for haste-server to run as.
+
+.. code:: bash
```
Comment (cjdh) [inline source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst]: Update command line format:

.. code-block:: console

    [server]$ adduser --system --home=/srv/haste-server --group haste-server

File: source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+
+    adduser --system --home=/srv/haste-server --group haste-server
+
+Now there is an ``haste-server`` user, change the permissions of
+/srv/haste-server so that it has access to the directory.
+
+.. code:: bash
```
Comment (cjdh) [inline source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst]: Update command line format:

.. code-block:: console

    [server]$ chown -R haste-server:haste-server /srv/haste-server

File: source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+
+    chown -R haste-server:haste-server /srv/haste-server
+
+Starting Hastebin
+~~~~~~~~~~~~~~~~~
+
+Finally start the service and set it to start at boot
+
+.. code:: bash
```
Comment (cjdh) [inline source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst]: Update command line format:

.. code-block:: console

    [server]$ systemctl enable haste-server
    [server]$ systemctl start haste-server

File: source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+
+    "port" : 8080,
+
+.. Note::
+
+    Read `How to configure access and security for DreamCompute instances
+    <https://help.dreamhost.com/hc/en-us/articles/215912838-How-to-configure-access-and-security-for-DreamCompute-instances>`__
+    for information on how to open port 8080 to traffic
```
Comment (cjdh) [inline source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst]: Add period at end of sentence.

File: source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+    WorkingDirectory=/srv/haste-server
+    Type=simple
+    User=haste-server
+    Group=haste-server
+    ExecStart=/usr/bin/node server.js
+
+    [Install]
+    WantedBy=multi-user.target
+
+Next we need to create the user for haste-server to run as.
```
Comment (cjdh) [inline source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst]: Remove 'we'. 

Next, create the user for haste-server to run as.

File: source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+
+.. code:: bash
+
+    adduser --system --home=/srv/haste-server --group haste-server
+
+Now there is an ``haste-server`` user, change the permissions of
```
Comment (cjdh) [inline source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst]: Change 'an' to 'a'. 

No there is a 'haste-server' user.

File: source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst [hunk 13]
```diff
@@ -0,0 +1,169 @@
...
+/srv/haste-server so that it has access to the directory.
+
+.. code:: bash
+
+    chown -R haste-server:haste-server /srv/haste-server
+
+Starting Hastebin
+~~~~~~~~~~~~~~~~~
+
+Finally start the service and set it to start at boot
```
Comment (cjdh) [inline source/dreamcompute/tutorials/deploy-hastebin-on-dreamcompute.rst]: Add semicolon at end of line.

**Pull request merged**